### PR TITLE
Add goal to storage

### PR DESCRIPTION
**Pull request by briantotheyanyan**
Just the first part of the storage.

### CODE REVIEW

File: assets/scripts/storage.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Saves new goals
```
Comment (Brian-Lam) [inline assets/scripts/storage.js]: This is raw JS, it should go in the build directory. Use Koala compiler to minify them and build into the assets directory.

Comment (Brian-Lam) [inline assets/scripts/storage.js]: Nit: Could you follow the function comment headers that the main files uses?

(sorry..)

File: assets/scripts/storage.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+function save() {
```
Comment (Brian-Lam) [inline assets/scripts/storage.js]: Let's change the function name, save() is pretty generic and if we make another save function later without thinking that could break some things

File: assets/scripts/storage.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	chrome.storage.sync.get('accomplishGoalsList', function(response){
+		var existingGoals = response.accomplishGoalsList;
+		if(typeof response.accomplishGoalsList === 'undefined') {
```
Comment (Brian-Lam) [inline assets/scripts/storage.js]: Nit: space between if and (

File: assets/scripts/storage.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+			existingGoals = [];
+		}
+		existingGoals.push({
+		'title': document.getElementsByName('newgoal-title')[0].value,
+		'begin': document.getElementsByName('newgoal-begin')[0].value,
+		'end': document.getElementsByName('newgoal-end')[0].value,
```
Comment (Brian-Lam) [inline assets/scripts/storage.js]: Let's just use jQuery for this

File: assets/scripts/storage.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+// Initialize method that checks if previous goals have been defined and
+// loads them if they have been
+function initialize() {
+	chrome.storage.sync.get('accomplishGoalsList', function(response){
+		if(typeof response.accomplishGoalsList !== 'undefined') {
+			response.accomplishGoalsList.forEach(function(goal) {
+
+				var titleTag = document.createElement('p');
```
Comment (Brian-Lam) [inline assets/scripts/storage.js]: Could we port all this over to jQuery for code style consistency?

File: assets/scripts/storage.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+				var descriptionTag = document.createElement('p');
+				var descriptionText = document.createTextNode("Description: " + goal.description);
+				descriptionTag.appendChild(descriptionText);
+				document.body.appendChild(descriptionTag);
+			});
+		}
+	});
+}
+
+document.addEventListener("DOMContentLoaded", function () {
```
Comment (Brian-Lam) [inline assets/scripts/storage.js]: Let's port this over the jQuery and run it at the top of the file, so someone who opens the file immediately knows what it does.

**Review by Brian-Lam — CHANGES_REQUESTED** [review]

**Comment by Brian-Lam** [comment]
Merge conflicts resolved

File: assets/scripts/storage.js (hunk 1 above)
Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

File: assets/scripts/storage.js (hunk 2 above)
Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

File: assets/scripts/storage.js (hunk 3 above)
Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

File: assets/scripts/storage.js (hunk 4 above)
Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

File: assets/scripts/storage.js (hunk 5 above)
Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

File: assets/scripts/storage.js (hunk 6 above)
Comment (briantotheyanyan) [reply assets/scripts/storage.js]: Resolved

**Review by Brian-Lam — APPROVED** [review]

**Pull request merged**